src/coffee/exportmapping.coffee

```diff
@@ -103,10 +104,16 @@ class ExportMapping
             row[index] = product[attribName][lang]
 
     if @header.has(CONS.HEADER_CATEGORY_ORDER_HINTS)
-      categoryIds = Object.keys product.categoryOrderHints
-      categoryOrderHints = _.map categoryIds, (categoryId) ->
-        return "#{categoryId}:#{product.categoryOrderHints[categoryId]}"
-      row[@header.toIndex CONS.HEADER_CATEGORY_ORDER_HINTS] = categoryOrderHints.join GLOBALS.DELIM_MULTI_VALUE
+      if product.categoryOrderHints?
+        categoryIds = Object.keys product.categoryOrderHints
+        categoryOrderHints = _.map categoryIds, (categoryId) =>
+          categoryIdentificator = categoryId
+          if @categoryOrderHintBy == 'externalId'
+            categoryIdentificator = @categoryService.id2externalId[categoryId]
+          return "#{categoryIdentificator}:#{product.categoryOrderHints[categoryId]}"
+        row[@header.toIndex CONS.HEADER_CATEGORY_ORDER_HINTS] = categoryOrderHints.join GLOBALS.DELIM_MULTI_VALUE
+      else
+        row[@header.toIndex CONS.HEADER_CATEGORY_ORDER_HINTS] = ''
```

Comment:
Currently, API is returning `categoryOrderHints` attribute even if it is empty so the export works as expected. It would make a difference only if we use this mapping function with an order draft which does not have to have this attribute but it is not very probable.